src/main/scala/wandou/avpath/Evaluator.scala

```diff
@@ -424,19 +452,20 @@ object Evaluator {
   private def evaluateObjectPredicate(expr: ObjPredSyntax, ctxs: List[Ctx]): List[Ctx] = {
     var res = List[Ctx]()
     ctxs foreach {
-      case currCtx @ Ctx(rec: IndexedRecord, _, _, _) =>
+      case currCtx @ Ctx(rec: IndexedRecord, _, _, _, _, _) =>
         evaluateExpr(expr.arg, currCtx) match {
           case true => res ::= currCtx
           case _    =>
         }
 
-      case Ctx(arr: java.util.Collection[_], schema, topLevelField, _) =>
+      case Ctx(arr: java.util.Collection[_], name, schema, topLevelField, path, _) =>
         getElementType(schema) foreach { elemType =>
           val values = arr.iterator
           var i = 0
           while (values.hasNext) {
             val value = values.next
-            val elemCtx = Ctx(value, elemType, topLevelField, Some(TargetArray(arr, i, schema)))
+            // not sure
```

Comment:
We are on a list, we name each sub element like the parent.
It should not be troublesome, but it's not right.